core/sis-referencing/src/test/java/org/apache/sis/geometry/CoordinateFormatTest.java

```diff
@@ -188,4 +193,76 @@ public void testParseGeographic4D() throws ParseException {
+        Locale locale = Locale.CANADA;
+        CoordinateFormat coordinateFormat = new CoordinateFormat(locale, null);
+        CoordinateFormat coordinateFormatTwo = coordinateFormat.clone();
+
+        assertNotSame(coordinateFormatTwo, coordinateFormat);
+    }
+
+    @Test
+    public void testParseAndFormatTaking2Arguments() throws IOException, ParseException {
+        CoordinateFormat coordinateFormat = new CoordinateFormat();
+        DirectPosition2D directPosition2D = new DirectPosition2D();
+        StringBuffer stringBuffer = new StringBuffer();
+        coordinateFormat.format(directPosition2D, stringBuffer);
+        ParsePosition parsePosition = new ParsePosition(0);
+        DirectPosition directPosition = coordinateFormat.parse(stringBuffer, parsePosition);
+
+        assertEquals(2, directPosition.getDimension());
+        assertEquals(2, directPosition.getCoordinate().length);
+        assertEquals(0.0, directPosition.getCoordinate()[0], 0.01);
+        assertEquals(0.0, directPosition.getCoordinate()[1], 0.01);
+    }
+
+    @Test
+    public void testParseAndSetDefaultCRSAndParse() throws ParseException {
+        CoordinateFormat coordinateFormat = new CoordinateFormat();
+        VerticalCRSMock verticalCRSMock = (VerticalCRSMock) VerticalCRSMock.BAROMETRIC_HEIGHT;
+        coordinateFormat.setDefaultCRS(verticalCRSMock);
+        ParsePosition parsePosition = new ParsePosition(4);
+        DirectPosition directPosition = coordinateFormat.parse("UTF-8", parsePosition);
+
+        assertEquals(1, directPosition.getDimension());
+        assertEquals(1, directPosition.getCoordinate().length);
+        assertEquals(8.0, directPosition.getCoordinate()[0], 0.01);
+    }
+
+    @Test
+    public void testApplyPatternReturningFalse() {
+        CoordinateFormat coordinateFormat = new CoordinateFormat();
+        Class<MessageFormat> clasz = MessageFormat.class;
+
+        assertFalse(coordinateFormat.applyPattern(clasz, "org.apache.sis.referencing.CommonCRS"));
+    }
+
+    @Test
+    public void testCreatesCoordinateFormatTaking2ArgumentsAndCallsGetPattern() {
+        Locale locale = Locale.UK;
+        ZoneOffset zoneOffset = ZoneOffset.MIN;
+        TimeZone timeZone = TimeZone.getTimeZone(zoneOffset);
+        CoordinateFormat coordinateFormat = new CoordinateFormat(locale, timeZone);
+        Class<Byte> clasz = Byte.class;
+
+        assertEquals("#,##0.###", coordinateFormat.getPattern(clasz));
+    }
+
+    @Test
+    public void testGetPatternReturningNull() {
+        CoordinateFormat coordinateFormat = new CoordinateFormat();
+        Class<XLink.Show> clasz = XLink.Show.class;
+
+        assertNull(coordinateFormat.getPattern(clasz));
+    }
+
+    @Test
+    public void testCreatesCoordinateFormatTakingNoArgumentsAndCallsSetSeparator() {
+        CoordinateFormat coordinateFormat = new CoordinateFormat();
+        coordinateFormat.setSeparator(" a - € ");
```

Comment:
Actually `setSeparator` was already used in other tests, but admittedly `getSeparator()` was not checked afterward. We can instead verifies `getSeparator()` in existing test methods.